### PR TITLE
Update release checklist: need to update docs and demo sites.

### PR DESCRIPTION
**Pull request by mauritsvanrees**

### CODE REVIEW

**Comment by mauritsvanrees** [comment]
No Jenkins jobs needed for this PR.

Just looking for feedback: if the involved people want a different text here to clarify what should be done, this is a good time to speak up.

**Review by fredvd — APPROVED** [review]
LGTM.  Seems something we could ‘officialise’ as release team tasks.

 Allthough if you look at it practically the release team is a virtual team where the release managers, ci-team, ai-team, docs, and training team members coordinate the subtasks from their team domain for every release.

File: release/checklist.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ See the [release schedule](https://plone.org/download/release-schedule).
 - [ ] Create tag of the `release/6.0-dev` branch, e.g. 6.0.0a1, and push to GitHub.
 - [ ] Make final release directory on dist.plone.org, with versions, requirements, constraints, changelog, release notes.
 - [ ] Update the "-latest" links on dist.plone.org, e.g. `ln -sfT 6.0.0a1 6.0-latest`
-- [ ] Notify Erico, perhaps on the `#install-and-deploy` Discord channel, that there is a new release.  He can create Docker images then.
+- [ ] Notify Erico Andrei and/or Fred van Dijk, for example the `#release-team` Discord channel, that there is a new release, so they can create Docker images then.  Sample changes: [`README.md`](https://github.com/plone/plone-backend/commit/0e3ce3c190677c04874ebed6b527a845cc075bca) and [`version.txt`](https://github.com/plone/plone-backend/commit/cee1e0bbb17ed24c64cd42342929a020346e1da1).
```
Comment (stevepiercy) [inline release/checklist.md]:
```suggestion
- [ ] Notify Érico Andrei and/or Fred van Dijk in the `#release-team` Discord channel that there is a new release. They will create Docker images. Examples of changes: [`README.md`](https://github.com/plone/plone-backend/commit/0e3ce3c190677c04874ebed6b527a845cc075bca) and [`version.txt`](https://github.com/plone/plone-backend/commit/cee1e0bbb17ed24c64cd42342929a020346e1da1).
```

File: release/checklist.md [hunk 2]
```diff
@@ -54,3 +54,5 @@ You probably want to wait until the Docker images are there, but don't wait long
 - [ ] Update the [release schedule](https://plone.org/download/release-schedule): note the new release, and say when the next release in this series is expected.
 - [ ] Edit the link on https://plone.org/download.
 - [ ] Announce on community.plone.org.
+- [ ] Update Plone version in documentation.  Ask Steve Piercy or do it yourself.  Here is a [sample PR](https://github.com/plone/documentation/pull/1492/files).
```
Comment (stevepiercy) [inline release/checklist.md]:
```suggestion
- [ ] In [`plone/documentation`'s `conf.py`](https://github.com/plone/documentation/blob/6-dev/docs/conf.py#L325-L326), update `PLONE_BACKEND_PATCH_VERSION` and `PLONE_BACKEND_MINOR_VERSION` as needed. Ask Steve Piercy or do it yourself. Here is a [sample PR](https://github.com/plone/documentation/pull/1492/files).
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Some grammar fixes and explicit details to include `PLONE_BACKEND_MINOR_VERSION` as well as needed.

**Review by stevepiercy — APPROVED** [review]
Thank you!

**Pull request merged**